.github/workflows/compiler_build_and_test_cpu_distributed.yml

```diff
@@ -45,81 +40,48 @@ jobs:
           echo "Request ID: ${{ inputs.request_id }}"
           echo "Matrix item: ${{ inputs.matrix_item }}"
 
-      # SSH private key is required as some dependencies are from private repos
-      - name: Set up SSH agent
-        uses: webfactory/ssh-agent@v0.7.0
-        with:
-          ssh-private-key: ${{ secrets.CONCRETE_CI_SSH_PRIVATE }}
+      - name: Instance cleanup
+        run: |
+          sudo rm -rf /home/ubuntu/actions-runner/_work/concrete/concrete/*
+          docker system prune -af
 
       - name: Fetch repository
         uses: actions/checkout@v3
         with:
+          fetch-depth: 0
           submodules: recursive
           token: ${{ secrets.CONCRETE_ACTIONS_TOKEN }}
 
-      - name: Setup rust toolchain for concrete-cpu
-        uses: ./.github/workflows/setup_rust_toolchain_for_concrete_cpu
+      - name: Set up home
+        # "Install rust" step require root user to have a HOME directory which is not set.
+        run: |
+          echo "HOME=/shared" >> "${GITHUB_ENV}"
 
-      - name: Create build dir
-        run: mkdir build
+      - name: Export specific variables (CPU)
+        if: ${{ !startswith(inputs.instance_type, 'p3.') }}
+        run: |
+          echo "CUDA_SUPPORT=OFF" >> "${GITHUB_ENV}"
+          echo "DATAFLOW_EXECUTION_ENABLED=ON" >> "${GITHUB_ENV}"
 
-      - name: Build compiler
-        uses: addnab/docker-run-action@v3
-        id: build-compiler
-        with:
-          registry: ghcr.io
-          image: ${{ env.DOCKER_IMAGE_TEST }}
-          username: ${{ secrets.GHCR_LOGIN }}
-          password: ${{ secrets.GHCR_PASSWORD }}
-          options: >-
-            -v ${{ github.workspace }}:/concrete
-            -v ${{ github.workspace }}/build:/build
-            -v ${{ github.workspace }}/wheels:/wheels
-            -v ${{ env.SSH_AUTH_SOCK }}:/ssh.socket
-            -e SSH_AUTH_SOCK=/ssh.socket
-            ${{ env.DOCKER_GPU_OPTION }}
-          shell: bash
-          run: |
-            rustup toolchain install nightly-2024-01-31
-            set -e
-            cd /concrete/compilers/concrete-compiler/compiler
-            rm -rf /build/*
-            make DATAFLOW_EXECUTION_ENABLED=ON CCACHE=ON Python3_EXECUTABLE=$PYTHON_EXEC BUILD_DIR=/build build-end-to-end-dataflow-tests
-            echo "Debug: ccache statistics (after the build):"
-            ccache -s
+      - name: Setup rust toolchain for concrete-cpu
+        uses: ./.github/workflows/setup_rust_toolchain_for_concrete_cpu
 
-      - name: Enable complete tests on push to main
-        if: github.ref == 'refs/heads/main'
-        run: echo "MINIMAL_TESTS=OFF" >> $GITHUB_ENV
+      - name: Build compiler benchmarks
```

Comment:
Why don't use the docker images to have caches?

Reply:
This is a persistent instance, so ccache will cover the compilation. 
The execution in any case cannot happen under docker as it would not easily be interoperable with SLURM (at least that would add a lot of complexity and it's hard to tell if the behaviour of code execution would still be representative of normal distributed execution).